lighthouse.go

```diff
@@ -358,6 +361,13 @@ func (lh *LightHouse) HandleRequest(rAddr *udpAddr, vpnIp uint32, p []byte, c *c
 			ans := NewUDPAddr(a.Ip, uint16(a.Port))
 			lh.AddRemote(n.Details.VpnIp, ans, false)
 		}
+		if lh.handshakeTrigger != nil {
```

Comment:
Would be nice to fix tests to avoid this nil check in production

Reply:
fixed in latest commit (this was actually unnecessary, a non-blocking send to a nil channel just skips the send).

I also added a test.